### PR TITLE
Use upstream shirou/gopsutil where possible

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Uses the upstream `shirou/gopsutil` where possible

### Motivation

Our fork `DataDog/gopsutil` has diverged significantly and isn't actively maintained.

### Additional Notes

I added comments about why I wasn't able to change certain usages. The most common reason is the usage of `process.AllProcesses()` which is something added to our fork.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/time_resolver.go
Comment (hmahmood) [inline pkg/network/tracer/time_resolver.go:1]: Going to remove this https://github.com/DataDog/datadog-agent/pull/22200

**Review by hmahmood — APPROVED** [review]

File: pkg/security/security_profile/activity_tree/process_node_snapshot.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ import (
 	"syscall"
 	"time"
 
+	// shirou/gopsutil uses different logic for getting the memory maps Path
```
Comment (paulcacheux) [inline pkg/security/security_profile/activity_tree/process_node_snapshot.go:23]: And there is a performance issue as well I believe (which was the reason we stayed with the legacy one)

**Review by paulcacheux — APPROVED** [review]

**Review by guyarb — APPROVED** [review]
looks good for USM owned files

File: cmd/manager/auto_exit_noprocess.go [hunk 2]
```diff
@@ -25,7 +25,20 @@ var (
...
+	if err != nil {
+		return nil, err
+	}
+
+	procs := make(processes)
+	for _, p := range ps {
+		name, err := p.Name()
+		if err != nil {
+			return nil, err
+		}
```
Comment (pgimalac) [inline cmd/manager/auto_exit_noprocess.go:38]: I think we might want to ignore the error (ie. just debug log it), if a process name cannot be fetched the most likely cause is that the process was stopped.
If returning an error is what `process.AllProcesses` did then feel free to ignore my comment

Comment (brycekahle) [reply cmd/manager/auto_exit_noprocess.go]: Updated to log

**Review by pgimalac — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

**Review by robertjli — APPROVED** [review]
Thank you!

**Review by rayz — APPROVED** [review]

**Pull request merged**